src/caffe/layers/data_layer.cpp

```diff
@@ -74,6 +74,7 @@ void DataLayer<Dtype>::Next() {
         << "Restarting data prefetching from start.";
     cursor_->SeekToFirst();
   }
+  offset_++;
```

Comment:
Shouldn't this reset when seeking back to the beginning or otherwise it will add up endlessly?